server/mdm/maintainedapps/ingest.go

```diff
@@ -0,0 +1,220 @@
+package maintainedapps
+
+import (
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/fleetdm/fleet/v4/pkg/fleethttp"
+	"github.com/fleetdm/fleet/v4/pkg/optjson"
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	kitlog "github.com/go-kit/log"
+	"golang.org/x/sync/errgroup"
+)
+
+//go:embed apps.json
+var appsJSON []byte
+
+type maintainedApp struct {
+	Identifier       string `json:"identifier"`
+	BundleIdentifier string `json:"bundle_identifier"`
+	InstallerFormat  string `json:"installer_format"`
+}
+
+const baseBrewAPIURL = "https://formulae.brew.sh/api/"
+
+// Refresh fetches the latest information about maintained apps from the brew
+// API and updates the Fleet database with the new information.
+func Refresh(ctx context.Context, ds fleet.Datastore, logger kitlog.Logger) error {
+	var apps []maintainedApp
+	if err := json.Unmarshal(appsJSON, &apps); err != nil {
+		return ctxerr.Wrap(ctx, err, "unmarshal embedded apps.json")
+	}
+
+	// allow mocking of the brew API for tests
+	baseURL := baseBrewAPIURL
+	if v := os.Getenv("FLEET_DEV_BREW_API_URL"); v != "" {
+		baseURL = v
+	}
+
+	i := ingester{
+		baseURL: baseURL,
+		ds:      ds,
+		logger:  logger,
+	}
+	return i.ingest(ctx, apps)
+}
+
+type ingester struct {
+	baseURL string
+	ds      fleet.Datastore
+	logger  kitlog.Logger
+}
+
+func (i ingester) ingest(ctx context.Context, apps []maintainedApp) error {
+	var g errgroup.Group
+
+	if !strings.HasSuffix(i.baseURL, "/") {
+		i.baseURL += "/"
+	}
+
+	client := fleethttp.NewClient(fleethttp.WithTimeout(10 * time.Second))
+
+	// run at most 3 concurrent requests to avoid overwhelming the brew API
+	g.SetLimit(3)
+	for _, app := range apps {
+		app := app // capture loop variable, not required in Go 1.23+
+		g.Go(func() error {
+			return i.ingestOne(ctx, app, client)
+		})
+	}
+	return ctxerr.Wrap(ctx, g.Wait(), "ingest apps")
+}
+
+func (i ingester) ingestOne(ctx context.Context, app maintainedApp, client *http.Client) error {
+	apiURL := fmt.Sprintf("%scask/%s.json", i.baseURL, app.Identifier)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, apiURL, nil)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "create http request")
+	}
+
+	res, err := client.Do(req)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "execute http request")
+	}
+	defer res.Body.Close()
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "read http response body")
+	}
+
+	switch res.StatusCode {
+	case http.StatusOK:
+		// success, go on
+	case http.StatusNotFound:
+		// TODO: delete the existing entry? do nothing and succeed? doing the latter for now.
```

Comment:
I think doing nothing makes sense in that case? In all likelihood, the existing entry (if any) would still work - its installer URL is not tied to brew.